core/src/main/java/org/apache/calcite/config/CalciteSystemProperty.java

```diff
@@ -420,6 +420,18 @@ public final class CalciteSystemProperty<T> {
   public static final CalciteSystemProperty<Integer> FUNCTION_LEVEL_CACHE_MAX_SIZE =
       intProperty("calcite.function.cache.maxSize", 0, v -> v >= 0);
 
+  /**
+   * The length of code before automatic method splitting is enabled
+   *
+   * <p>A value of <code>0</code>, the default, disables automatic method splitting.
+   *
+   * <p>Some queries can generate methods exceeding the JVM limit of 4000 characters per method.
```

Comment:
Can you include a citation for your statement "JVM limit of 4000 characters per method"?

Java 7 had a limit [65534 bytes](https://docs.oracle.com/javase/specs/jvms/se7/html/jvms-4.html#jvms-4.7.3) (bytecode size). When did that change?

Reply:
I got this 4K from Flink's documentation on the configuration option relating to this:
https://github.com/apache/flink/blob/4101316ef37d3a7082bff57cb732f4e9e0349d09/flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/config/TableConfigOptions.java#L225

So it's not exactly that the JVM has a limit of 4K, it's that the default JIT allows 8K only. Let me find a source for this.

Reply:
It's specified here in OpenJDK (other vendors use the same limit): https://github.com/openjdk/jdk11u-dev/blob/a754a3d8972abc67aee768d31a2dc2e8214274cf/src/hotspot/share/runtime/globals.hpp#L2391

It's considered a developer option though, rather than an end-user option, so documentation is limited.